### DISCUSSION TITLE
Supporting multiple wallet connections on a single web dApp.

### DISCUSSION THREAD

**Discussion by jonandgon** [WalletConnect/walletconnect-monorepo#2479] [Developer Support]
I'm creating a bridging dApp that sends funds from an L1 wallet to an L2 wallet. I'd like to use walletconnect/web3modal to connect both wallets to my dApp. However, it's not currently possible to do so since the info from one modal overwrites the info of another in the local storage. 

Is there a way to make this work? If not, are there plans to support multiple connections?

**Comment by finessevanes** [comment]
@jonandgon just so I understand, you want to connect two different accounts that are in two different wallets?

**Reply by jonandgon** [reply]
yes exactly

**Reply by finessevanes** [reply]
I don't believe this feature exists ATM, but let me check w/ our team!

**Reply by jonandgon** [reply]
Thanks so much! I'm building something similar to this fyi. https://starkgate.starknet.io/

**Reply by finessevanes** [reply]
It doesn't exist yet, but there has been action to start the initiative. Would you like to stay in the know and perhaps pilot the feature?

**Reply by jonandgon** [reply]
Yes! I'd love that. We've got an immediate need for it so we'll be happy to be the guinea pigs.

**Reply by finessevanes** [reply]
Would you mind sending me an email at vanes@walletconnect.com ?

**Reply by app-matt** [reply]
*Bump* - Just wondering if this was ever addressed?
